### ISSUE TITLE
Vehicleid for existing users not showing up

### ISSUE THREAD

**Issue by jeannovak** [OWASP/crAPI#75]
When accessing the forum, a GET request to /community/api/v2/community/posts/recent is done to receive information about posts and comments. There are 3 existing posts made by the 3 default users, however there is no information about their vehicles.

![image](https://user-images.githubusercontent.com/42330295/172621717-a02f05d4-407a-4c70-84c9-db8ec9faeccc.png)

Which is honestly not a big deal, you can simply create two users, post something and use their vehicle guids for the challenges. However, I think it would be nice to already have some pre-built vehicleids so the challenge progress feels more natural. I think the vehicles are already created because there is mechanic reports for the default users, so I'm not sure if this is the expected behaviour.

Both last stable and development versions have this behaviour. Environment: Macbook M1, macOS Monterey 12.4, Docker 4.7.1 (77678).

**Comment by Ph4t3** [comment]
I was not able to reproduce this bug. Can you give the logs of crapi-identity container? Run the command
```docker logs crapi-identity```

**Comment by mathew-jose** [comment]
Did this error occur while running the `/community/api/v2/community/posts/recent` API request as the errors indicated in the log files pertains to setting a token. I was not able to replicate this error while pulling the pre-built images from dockerhub.

**Comment by jeannovak** [comment]
@mathew-jose yes I am running `/community/api/v2/community/posts/recent` API request because its where I can see if the posts were populated correctly after bringing the stack up, but the endpoint itself seems to not be related to the issue. 

I updated docker for testing purposes and tested multiple times while removing all volumes and even trying download images again. In any case, I have been able to replicate the issue quite consistently.

I though it was weird that the e-mail would show up but not the other data like nickname, so I looked at the code and noticed that posts, e-mails and their insertion in mongodb are defined in `services/community/api/seed/seeder.go`, while the user information (name, e-mail and tepehone number) comes from `services/identity/src/main/java/com/crapi/config/InitialDataConfig.java`, which will only run based on call below (line 83).

```
    @EventListener
    public void setup(ApplicationReadyEvent event){

        addLocation();
        addVehicleModel();
        addUser();
    }
```
This seems to wait for the application to be ready before creating the users.

In one of my last tests a scenario, Adam had a nickname in addition to e-mail, while the others had only e-mails:

```
HTTP/1.1 200 OK
Server: openresty/1.17.8.2
Date: Sat, 18 Jun 2022 16:12:02 GMT
Content-Type: application/json
Connection: close
Access-Control-Allow-Headers: Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization
Access-Control-Allow-Methods: POST, GET, OPTIONS, PUT, DELETE
Access-Control-Allow-Origin: *
Content-Length: 827

[
  {
    "id": "JHekNigM85BG8jrqGJtrvc",
    "title": "Title 3",
    "content": "Hello world 3",
    "author": {
      "nickname": "",
      "email": "robot001@example.com",
      "vehicleid": "",
      "profile_pic_url": "",
      "created_at": "2022-06-18T16:09:43.865Z"
    },
    "comments": [],
    "authorid": 0,
    "CreatedAt": "2022-06-18T16:09:43.865Z"
  },
  {
    "id": "hbBoQSusURQvHPempqWzJn",
    "title": "Title 2",
    "content": "Hello world 2",
    "author": {
      "nickname": "",
      "email": "pogba006@example.com",
      "vehicleid": "",
      "profile_pic_url": "",
      "created_at": "2022-06-18T16:09:43.863Z"
    },
    "comments": [],
    "authorid": 0,
    "CreatedAt": "2022-06-18T16:09:43.863Z"
  },
  {
    "id": "hjAC2Aa5W7c6abwG4UbqyR",
    "title": "Title 1",
    "content": "Hello world 1",
    "author": {
      "nickname": "Adam",
      "email": "adam007@example.com",
      "vehicleid": "",
      "profile_pic_url": "",
      "created_at": "2022-06-18T16:09:43.844Z"
    },
    "comments": [],
    "authorid": 1,
    "CreatedAt": "2022-06-18T16:09:43.844Z"
  }
]
```
So I decided to take a look at the database. It had the 3 users, Adam, Dogba and Robot, including their telephone number, as well as three vehicles. So the issue doesn't seem to be related to InitialDataConfig.java:

postgresdb user_details:
![image](https://user-images.githubusercontent.com/42330295/174447648-f98b80c4-3761-4581-a015-9372faee466d.png)
 
postgresdb vehicle_details:
![image](https://user-images.githubusercontent.com/42330295/174447894-74c296c4-ac95-4cd5-859c-03eaa0527c9d.png)

However, the mongodb crapi.post collection seems empty:

```
switched to db crapi
crapi> db.post.find().pretty()
[
  {
    _id: ObjectId("62adf8c7d3ff5daa6a1e239a"),
    id: 'hjAC2Aa5W7c6abwG4UbqyR',
    title: 'Title 1',
    content: 'Hello world 1',
    author: {
      nickname: 'Adam',
      email: 'adam007@example.com',
      vehicleid: '',
      picurl: '',
      createdat: ISODate("2022-06-18T16:09:43.844Z")
    },
    comments: [],
    authorid: Long("1"),
    createdat: ISODate("2022-06-18T16:09:43.844Z")
  },
  {
    _id: ObjectId("62adf8c7d3ff5daa6a1e239b"),
    id: 'hbBoQSusURQvHPempqWzJn',
    title: 'Title 2',
    content: 'Hello world 2',
    author: {
      nickname: '',
      email: 'pogba006@example.com',
      vehicleid: '',
      picurl: '',
      createdat: ISODate("2022-06-18T16:09:43.863Z")
    },
    comments: [],
    authorid: Long("0"),
    createdat: ISODate("2022-06-18T16:09:43.863Z")
  },
  {
    _id: ObjectId("62adf8c7d3ff5daa6a1e239c"),
    id: 'JHekNigM85BG8jrqGJtrvc',
    title: 'Title 3',
    content: 'Hello world 3',
    author: {
      nickname: '',
      email: 'robot001@example.com',
      vehicleid: '',
      picurl: '',
      createdat: ISODate("2022-06-18T16:09:43.865Z")
    },
    comments: [],
    authorid: Long("0"),
    createdat: ISODate("2022-06-18T16:09:43.865Z")
  }
]
```

Is it possible that data is being added to mongodb collection before the addition of Adam, Robot and Pogba to postgres? This would explain why the postgresdb seems to contain all info but mongodb lacks data, as well as a possible explanation for the scenario above where only Adam has a nickname. I apologize if this has nothing to do with the issue as I'm not familiar with java.

**Comment by mikeacjones** [comment]
Can't reproduce this issue either. Have you attempted to add a vehicle after signup manually via MailHog? I just scripted out some tests, and all posts are showing the vehicle ID.

**Comment by johanlindfors** [comment]
I am experiencing this issue as well today, on a Ubuntu 22.04 device. No vehicleId's appear in the response for recent posts.
